### DISCUSSION TITLE
Mikado 0.8.0

### DISCUSSION THREAD

**Discussion by ts-thomas** [nextapps-de/mikado#49] [Announcements]
Until the end of this year I plan to make a full refactoring of Mikado including a lot of quality improvements:

- improve template compiler
- improve template factory
- support for text nodes within templates (non-element childs)
- improve template pooling
- simplify Mikado options
- simplify rendering modes
- improve naming convention
- support for auto-mount + auto-unmount
- built-in sanitizer for HTML contents
- improve reactive paradigm and template capabilities
- improve code structure, modularization and build system
- rewrite documentation
- improve benchmarks and test suite

Do you see any issues which is missing here? Please comment below ...

**Comment by Dan-Do** [comment]
These are all great improvements. I have one concern that, with current version, mikado only hides the element
`<div if="false">foo</div>` is rendered as `<div hidden>foo</div>`.
Can we have an option to remove it completely from the DOM when template is rendered?

**Reply by ts-thomas** [reply]
Yes that would be much better than hiding. This is a very difficult task because of the internal render-tree which could then vary between different components when rendered. It is solvable but I need some time for this to find a satisfying solution.

**Reply by ts-thomas** [reply]
Conditional template structures are now supported in v0.8.x

**Comment by bloody-ux** [comment]
What's the plan for Mikado? It's not updated for months.

**Comment by cayolblake** [comment]
@ts-thomas 

Any updates on the progress/state of 0.8.x ?

Is the project still alive and safe for use in product given it wasn't updated since long ?

**Reply by Dan-Do** [reply]
Yes, I am using it daily for my personal site.

**Comment by sirinath** [comment]
Wondering what the progress is? When can we expect a release?

**Comment by niutech** [comment]
Please add progressive enhancement like [Petite Vue](https://github.com/vuejs/petite-vue), so that we could spark Mikado on top of existing static HTML to add small interactions (like tabs, sliders, carousels, lightboxes etc.)

**Reply by ts-thomas** [reply]
Mikado v0.8.x is providing hydration
